src/model.ts

```diff
@@ -1379,6 +1420,231 @@ export class GitExtension implements IGitExtension {
+   * 
+   */
+  async stash_pop(index?: number): Promise<void>{
+
+    try{ 
+      const path = await this._getPathRepository();
+      
+      const stashFiles = this._stash[index].files;
+  
+      await this._taskHandler.execute<void>('git:stash:pop', async() => {
+        await requestAPI(URLExt.join(path, 'stash_pop'), 'POST', 
+          index !== undefined ? {index} : undefined
+        );
+      })
+  
+  
+      await this.refresh();
+  
+      stashFiles.forEach(file=>{
+        this._refreshAffectedFile(file)
+      })
+  
+    } catch (error) {
+      console.error('Failed to pop stash', error);
+    }
+ 
+    await this.refreshStash();
+
+  }
+  
+  async stash_apply(index?: number): Promise<void> {
+    const path = await this._getPathRepository();
+    try {
+
+      // Grab the list 
+      const stashFiles = this._stash[index].files;
+      // if stashFiles is empty, throw an error
+
+      await this._taskHandler.execute<void>('git:stash:apply', async() => {
+        await requestAPI(URLExt.join(path, 'stash_apply'), 'POST', 
+          index !== undefined ? {index} : {index:0}
+        );
+      })
+
+      await this.refresh();
+
+      stashFiles.forEach(file=>{
+        this._refreshAffectedFile(file)
+      })
+
+    } catch (error) {
+      console.error('Failed to apply stash', error);
+
+      // should i spin dialog box? 
+    }
+
+    await this.refreshStash();
+  }
+
+  /**
+   * Checks the stash list, and sets the stash property.
+   * 
+   * @returns promise which resolves to an array of stashes
+   * @throws {Git.NotInRepository} If the current path is not a Git repository
+   * @throws {Git.GitResponseError} If the server response is not ok
+   * @throws {ServerConnection.NetworkError} If the request cannot be made
+   */
+  async refreshStash(): Promise<void> {
+    let path: string;
+
+    try {
+      path = await this._getPathRepository();
+    } catch (error) {
+      this._clearStatus();
+      if (!(error instanceof Git.NotInRepository)) {
+        throw error;
+      }
+      return;
+    }
+
+    // Get the entire stash list
+    try {
+      const stashListData = await this._taskHandler.execute<Git.IStashListResult>(
+        'git:refresh:stash',
+        async () => {
+          return await requestAPI<Git.IStashListResult>(
+            URLExt.join(path, 'stash_list'),
+            'POST'
+          );
+        }
+      );
+
+      
+      // Contains the raw message
+      const stashMsgList = stashListData.message.split('\n').slice(0,-1)
+      
+      const stashList: Git.IStashEntry[] = []
+      for (let index in stashMsgList) {
+          
+          const stashInfo = stashMsgList[index].split(':')
+
+          const branchName = stashInfo[1].match(/WIP on (.*)/) ? stashInfo[1].match(/WIP on (.*)/)[1] : stashInfo[1].split(" ")[2];
+
+          const stashMsg = stashInfo[2].replace(/^\s+/, '');
+
+          stashList.push({
+              index: parseInt(index),
```

Comment:
You need to specify the radix because the default is not 10.
```suggestion
              index: parseInt(index, 10),
```